src/hvm.rs

```diff
@@ -3623,21 +3623,38 @@ pub fn name_to_u128(name: &str) -> u128 {
   let mut num: u128 = 0;
   for (i, chr) in name.chars().enumerate() {
     debug_assert!(i < 20, "Name too big: `{}`.", name);
-    if chr == '.' {
-      num = num * 64 + 0;
-    } else if chr >= '0' && chr <= '9' {
-      num = num * 64 + 1 + chr as u128 - '0' as u128;
-    } else if chr >= 'A' && chr <= 'Z' {
-      num = num * 64 + 11 + chr as u128 - 'A' as u128;
-    } else if chr >= 'a' && chr <= 'z' {
-      num = num * 64 + 37 + chr as u128 - 'a' as u128;
-    } else if chr == '_' {
-      num = num * 64 + 63;
-    }
+    num = (num << 6) + char_to_u128(chr);
   }
   return num;
 }
 
+pub const fn char_to_u128(chr: char) -> u128 {
+  match chr {
+    '.'       => 0,
+    '0'..='9' =>  1 + chr as u128 - '0' as u128,
+    'A'..='Z' => 11 + chr as u128 - 'A' as u128,
+    'a'..='z' => 37 + chr as u128 - 'a' as u128,
+    '_'       => 63,
+    _         => panic!("Invalid name character."),
+  }
+}
+
+pub const fn const_name_to_u128(name: &str) -> u128 {
+  let name = name.as_bytes();
+  let last = name.len() as isize - 1;
+  let mut res: u128 = 0;
+  macro_rules! set {
+    ( $I:expr ) => {
+      let i = $I; let pos = last-i; if pos >= 0 { res = res | (char_to_u128(name[pos as usize] as char) << (i*6)); }
```

Comment:
This is much less readable. I don't think the benefit of the consts is worth making this function less clear.

Reply:
I just did some refactoring, but i'm not satisfied with it yet.

Reply:
Btw, the non const function still exists.